watertap/unit_models/selective_oil_permeation.py

```diff
@@ -0,0 +1,669 @@
+    **MomentumBalanceType.pressureTotal** - single pressure balance for material,
+    **MomentumBalanceType.pressurePhase** - pressure balances for each phase,
+    **MomentumBalanceType.momentumTotal** - single momentum balance for material,
+    **MomentumBalanceType.momentumPhase** - momentum balances for each phase.}""",
+        ),
+    )
+    CONFIG.declare(
+        "has_pressure_change",
+        ConfigValue(
+            default=True,
+            domain=In([True, False]),
+            description="Pressure change term construction flag",
+            doc="""Indicates whether terms for pressure change should be
+    constructed,
+    **default** - True.
+    **Valid values:** {
+    **True** - include pressure change terms,
+    **False** - exclude pressure change terms.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        # Add unit parameters
+        self.pore_diameter = Param(
+            mutable=True,
+            initialize=4e-8,
+            units=units_meta("length"),
+            doc="Average membrane pore diameter",
+        )
+
+        self.porosity = Param(
+            mutable=True,
+            initialize=0.4,
+            units=units.dimensionless,
+            doc="Membrane porosity",
+        )
+
+        self.membrane_thickness = Param(
+            mutable=True,
+            initialize=4e-5,
+            units=units_meta("length"),
+            doc="Membrane thickness",
+        )
+
+        self.permeability_constant = Param(
+            mutable=True,
+            initialize=150,
+            units=units.dimensionless,
+            doc="Permeability constant",
+        )
+
+        self.module_diameter = Param(
+            mutable=True,
+            initialize=0.064,
+            units=units_meta("length"),
+            doc="Module diameter",
```

Comment:
All of these can be variables that get fixed by the user after creating an instance of the unit model